File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
// Commulative Sum

/*

[1,1,1] 2


*/

#include <iostream>
using namespace std;

class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
        int sum = 0;
        int count = 0;
        
        unordered_map<int,int> seen = {{0,1}};
        
        for(int n: nums){
            sum +=n;
            count += seen[sum-k];
            seen[sum]++;
        }
        
        return count;
        
    }
};

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
## Commulative Sum 
import collections

class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        
        seen = collections.Counter({0:1})
        sum = 0
        count = 0
        
        for _, x in enumerate(nums):
            sum += x
            count += seen[sum-k]
            seen[sum] +=1 
        return count
        